subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionSkipCachedStateIntegrationTest.groovy

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution
+
+class InstantExecutionSkipCachedStateIntegrationTest extends AbstractInstantExecutionIntegrationTest {
+
+    def "skip reading cached state on --refresh-dependencies"() {
+
+        given:
+        buildFile << """
+            class MyTask extends DefaultTask {
+                @TaskAction def action() {
+                    println("foo")
+                }
+            }
+            tasks.register("myTask", MyTask)
+        """
+
+        when:
+        instantRun "myTask"
+
+        then:
+        outputContains("foo")
+
+        when:
+        instantRun "myTask"
+
+        then:
+        outputContains("foo")
+
+        when:
+        buildFile.text = buildFile.text.replace("foo", "bar")
+
+        and:
+        instantRun "myTask"
+
+        then:
+        outputContains("foo")
+
+        when:
+        instantRun "myTask", "--refresh-dependencies"
+
+        then:
+        outputContains("bar")
+
+        when:
+        instantRun "myTask"
```

Comment:
⭕️ Assert the cached work graph was reused here?

Reply:
What about we introduce build operations for saving and loading the work graph so we can assert on them, have them record timings and be shown in e.g. chrome traces? I'd do that in a subsequent PR though, reworking all existing tests to leverage them. How does that sound?

Reply:
Sounds great.